client/pages/topsecret.jsx

```diff
@@ -1,43 +1,173 @@
 import React from 'react';
 import * as ReactBS from 'react-bootstrap';
-import {Navigation, Router} from 'react-router';
+import {Navigation} from 'react-router';
+import _ from 'lodash';
 
 const HomePage = React.createClass({
+
   mixins : [Navigation],
+
+  getInitialState() {
+    return {
+      initialBankSavings : 4294967296.69, initialExpenses : 0, tools : [],
+      addlbl : '', skuin : '', typein : '',
+      namein : '', costin : '', costQ : '', nameQ : '', searchdebug : '',
+      costfilter : '>',
+    };
+  },
+
   componentDidMount() {
     var {userid} = this.context.router.getCurrentQuery();
-    if(userid != 1)
-    {
-      //router.transitionTo('/');
+    if (userid != 1) {
+      // router.transitionTo('/');
       this.replaceWith('/');
-      console.log(this.context);
     }
 
+
   },
+  handleChange(key, event) {
+      this.setState({[key] : event.currentTarget.value});
+    },
+
+  handleAddClick(event) {
+      const doesExist = _.find(this.state.tools, { sku : this.state.skuin });
+      if (this.state.skuin == '' || this.state.namein == ''
+      || this.state.typein == '' || this.state.costin == '') {
+        this.setState({addlbl :
+         'You have left an input field blank, please fill out all input fields.'});
+        return;
+      }
+
+      this.setState({addlbl : 'The tool you enterred already exists in the armory!'});
+      if (!isNaN(this.state.costin)) {
+        if (!doesExist || this.state.tools.length == 0) {
+           // array
+          var temparray = this.state.tools.slice();
+          var newexp = Number(this.state.initialExpenses) + Number(this.state.costin);
+          temparray.push({sku : this.state.skuin,
+            name : this.state.namein, type : this.state.typein, cost : Number(this.state.costin)});
+           // update labels
+          this.setState({initialExpenses : newexp,
+             tools : temparray,
+             addlbl : 'The item with SKU: ' + this.state.skuin + ', Type: ' + this.state.typein
+              + ', Name: ' + this.state.namein + ', and Cost: $' + this.state.costin
+              + ' has been added to the armory.'},
+          );
+
+
+        } else {
+          this.setState({addlbl : 'The tool you enterred already exists in the armory!'});
+        }} else {
+        this.setState({addlbl :
+         'You have enterred an invalid cost; enter only numbers (XXXX.XX or XXXX)'});
+      } },
+
+  itemFinder(tool) {
```

Comment:
I think we can reduce the amount of defined variables and boolean checks by grouping similar logic together here. For instance c1, c2, and c3, all mean that the cost filter matched somehow so we can probably group those into one variable. The other check is an item name match. 

Since blankn and blankc aren't manipulated or changed (they are just copies of state) we can get rid of those altogether. 

```

  itemFinder(tool) {
    let itemNameMatch, costMatch;

    if (tool.name.indexOf(this.state.nameQ) > -1) {
      itemNameMatch = true;
    }

    if (! (this.state.nameQ || this.state.costQ)) {
      return false; // user didn't apply any filters
    }

    if ((this.state.costfilter === '=' && tool.cost === Number(this.state.costQ)) ||
      (this.state.costfilter === '>' && tool.cost > Number(this.state.costQ)) ||
      (this.state.costfilter === '<' && tool.cost < Number(this.state.costQ))) {
      costMatch = true;
    }

    /* Return true if
       If user specified a both filters and there were matches
       If user specified just cost filter and there was a match
       If user specified just name filter and there was a match */
    if (
      (! this.state.nameQ && costMatch) ||
      (! this.state.costQ && itemNameMatch) ||
      (itemNameMatch && costMatch)
    ) {
      return true;
    }

    return false;
  },

```